### ISSUE TITLE
[Bug] Not Recognisng Artwork

### ISSUE THREAD

**Issue by Qwerty-Space** [Taiko2k/TauonMusicBox#413]
For some reason Tauon would not recognise that a file had embedded album art.  After re-encoding the image, it started to recognise it.

Original image:
![folder](https://user-images.githubusercontent.com/6967592/96724642-f83ac780-13a7-11eb-98bb-d692d0e75e0a.png)

**Comment by Taiko2k** [comment]
I would really need the original file to investigate into this.

By re-encode, do you mean re-embed?

**Comment by Scotty-Trees** [comment]
@Qwerty-Space I had a similar issue that happened to 2-3 of my albums. If you can get it, download Kid3 tag editor ==> [flatpak version](https://flathub.org/apps/details/org.kde.kid3) and you can select and delete whatever album artwork is attached to it. Just make sure to save the file before you exit otherwise the changes won't take affect of course. Also, if you delete the album artwork, if you have a different file to use, you can just drag and drop it into the app to replace it and then save it. It's the onyl fix for something like this, it's not really a Tauon issue, when it's more in line with the integrity of the music files. Generally though, this problem is pretty rare, but this is one way to fix it. If you need more help, to fix it with my method, just ping me.

**Comment by Qwerty-Space** [comment]
> I would really need the original file to investigate into this.
> 
> By re-encode, do you mean re-embed?

Yes. I can try putting the old album art on it again to see if it breaks

**Comment by Qwerty-Space** [comment]
> @Qwerty-Space I had a similar issue that happened to 2-3 of my albums. If you can get it, download Kid3 tag editor ==> [flatpak version](https://flathub.org/apps/details/org.kde.kid3) and you can select and delete whatever album artwork is attached to it. Just make sure to save the file before you exit otherwise the changes won't take affect of course. Also, if you delete the album artwork, if you have a different file to use, you can just drag and drop it into the app to replace it and then save it. It's the onyl fix for something like this, it's not really a Tauon issue, when it's more in line with the integrity of the music files. Generally though, this problem is pretty rare, but this is one way to fix it. If you need more help, to fix it with my method, just ping me. 

I stated in the original post that I fixed it by changing the album artwork on the file

**Comment by Qwerty-Space** [comment]
Update:  It's clearly an issue with the embedded image.  Adding it back broke it again.
![image](https://user-images.githubusercontent.com/6967592/97020019-f3f9e000-1548-11eb-8289-ca37fba8d3db.png)

I'll send the audio file on Discord

**Comment by Taiko2k** [comment]
Indeed looks like you were right in it being related to the image format, being 32bit. Tauon uses Pillow for image processing which doesn't seem to support that (yet). Ref https://github.com/python-pillow/Pillow/issues/1888. So, there's pretty much no workaround for this until that issue is solved.

**Comment by Qwerty-Space** [comment]
@Taiko2k I see.  One of us should open up an issue there, or for now maybe add a special case for 32bit artwork?  Maybe display a placeholder that outlines the issue?

**Comment by Taiko2k** [comment]
I linked to the existing issue for it in my above comment.

>  Maybe display a placeholder that outlines the issue?

Seems that it gives a generic error so I cant really detect specifically that the cause of failure was the image being 32bit

**Comment by C0rn3j** [comment]
Pillow now has a hanging PR for this with requests for  testing/code review - https://github.com/python-pillow/Pillow/issues/1888#issuecomment-2180370073
